src/V3SplitVar.cpp

```diff
@@ -0,0 +1,719 @@
+        std::vector<SplitNewVar> plan;
+        std::vector<std::pair<int, bool> > points;  // <bit location, is end>
+        points.reserve(m_lhs.size() * 2 + 2);  // 2 points will be added per one Entry
+        for (const_iterator it = m_lhs.begin(), itend = m_lhs.end(); it != itend; ++it) {
+            points.push_back(std::make_pair(it->lsb(), false));     // start of a region
+            points.push_back(std::make_pair(it->msb() + 1, true));  // end of a region
+        }
+        if (skipUnused && !m_rhs.empty()) {  // range to be read must be kept, so add points here
+            int lsb = m_basicp->msb() + 1, msb = m_basicp->lsb() - 1;
+            for (size_t i = 0; i < m_rhs.size(); ++i) {
+                lsb = std::min(lsb, m_rhs[i].lsb());
+                msb = std::max(msb, m_rhs[i].msb());
+            }
+            UASSERT_OBJ(lsb <= msb, m_basicp, "lsb:" << lsb << " msb:" << msb << " are wrong");
+            points.push_back(std::make_pair(lsb, false));
+            points.push_back(std::make_pair(msb + 1, true));
+        }
+        if (!skipUnused) {  // all bits are necessary
+            points.push_back(std::make_pair(m_basicp->lsb(), false));
+            points.push_back(std::make_pair(m_basicp->msb() + 1, true));
+        }
+        std::sort(points.begin(), points.end(), SortByFirst());
+
+        // scan the sorted points and sub bitfields
+        int refcount = 0;
+        for (size_t i = 0; i + 1 < points.size(); ++i) {
+            const int bitwidth = points[i + 1].first - points[i].first;
+            if (points[i].second)
+                --refcount;  // end of a region
+            else
+                ++refcount;  // start of a region
+            UASSERT(refcount >= 0, "refcounut must not be negative");
+            if (bitwidth == 0 || refcount == 0) continue;  // vacant region
+            plan.push_back(SplitNewVar(points[i].first, bitwidth));
+        }
+
+        return plan;
+    }
+};
+
+class SplitPackedVarVisitor : public AstNVisitor {
+    AstNetlist* m_netp;
+    AstNodeModule* m_modp;  // current module (just for log)
+    int m_numSplit;         // total number of split variable
+    bool m_isLhs;           // true when traversing LHS of assignment
+    // key:variable to be split. value:location where the variable is referenced.
+    vl_unordered_map<AstVar*, PackedVarRef> m_refs;
+    virtual void visit(AstNode* nodep) VL_OVERRIDE { iterateChildren(nodep); }
+    virtual void visit(AstAssign* nodep) VL_OVERRIDE {
+        UASSERT_OBJ(m_isLhs == false, nodep, "unexpected nested assign");
+        m_isLhs = true;
+        iterate(nodep->lhsp());
+        m_isLhs = false;
+        iterate(nodep->rhsp());
+    }
+    virtual void visit(AstAssignW* nodep) VL_OVERRIDE {
+        UASSERT_OBJ(m_isLhs == false, nodep, "unexpected nested assign");
+        m_isLhs = true;
+        iterate(nodep->lhsp());
+        m_isLhs = false;
+        iterate(nodep->rhsp());
+    }
```

Comment:
Seems suspicious you do Assign and AssignW but not NodeAssign, e.g. you don't do AssignDly's etc.